datafusion/proto/src/logical_plan/from_proto.rs

```diff
@@ -655,29 +656,36 @@ impl TryFrom<&protobuf::ScalarValue> for ScalarValue {
                 let schema: Schema = if let Some(schema_ref) = schema {
                     schema_ref.try_into()?
                 } else {
-                    return Err(Error::General("Unexpected schema".to_string()));
+                    return Err(Error::General(
```

Comment:
Should we return some of Datafusion Errors instead? Maybe IoError or InternalError?

Reply:
I am not sure why the from_proto and to_proto modules have their own error types and don't use `DataFusionError` (or even wrap `DataFusionError`)

It does seem like that would be an improvement. Maybe I can file a follow on ticket to do so.